paper.tex

```diff
@@ -286,20 +289,109 @@ \subsection*{Human Data Model}
+mechanism for specifying default values that are inherited hierarchically.
+For values that repeat
+across fields, defaults may be used to implicitly assign default values to
+fields of the given type.  A default is superseded by an explicitly provided
+value if given. Size values are inherited naturally following the
+progression of time. For example, if an epoch \texttt{start\_size} is not
+provided (neither directly, nor via a defaults section),
+it is assumed to be equal to the \texttt{end\_size} of the previous
+epoch. This also means that the first epoch of each population must specify the
+initial size (or it must be provided in a defaults section).
+
+Avoiding redundancy in this way reduces the cognitive load on readers,
+by highlighting necessary parameters which may be otherwise be obscured.
+It is not necessary---or indeed recommended---that all models are expressed
+in a maximally concise form, and we wholeheartedly endorse
+the explicit statement of parameters where it increases model legibility.
 
+\subsection*{Parsers}
+
+While the HDM is designed for human readability and
+conciseness, the underlying data model suitable for software implementation
+(the Machine Data Model, or MDM)
+is redundant and exhaustive.
+Translation from the HDM to the MDM
+requires resolving hierarchically-defined default values and verifying
+relationships between populations and the validity of specified parameter
+values.
+Because this translation and validation requires significant
+programming effort, we define a standard software entity as part of the
+specification to perform this task (the parser),
+which is intended to be shared by programs that support Demes as input.
+The Demes specification precisely defines the required behavior of parsers,
+and we provide a reference implementation written in Python to resolve any
+potential ambiguities, as well as an extensive test suite of examples and the
+expected outputs. In addition, we have high-quality parser implementations in
+the Python and C languages (Table~\ref{tab:software})
+providing a solid foundation for the software ecosystem.
+By maintaining high-quality Demes parsers available as libraries,
+we ensure consistency across simulation and inference
+software.
+Having common parsers also benefits users by providing
+consistent and informative error messages for missing values or
+issues in formatting.
+
+\subsection*{Scope of the specification}
+
+A primary design goal of Demes is to provide a means of
+unambiguously communicating the results of demographic
+model inferences to population genetic simulators.
+Since demography is defined in terms of groups of individuals
+and these groupings are influenced by genetics,
+it is difficult to find a simple definition that separates
+the two. Thus, we have attempted to be pragmatic,
+limiting the features that we include in Demes to
+those that are (or can reasonably be expected to)
```

Comment:
Wording here seems a bit off...

"(or can reasonably be expected to be)"

Reply:
Yes, a bit jumbled - I've dropped the paranthetical comment.